pkg/catalog/catalog.go

```diff
@@ -21,6 +22,17 @@ var DevfileRegistries = []string{
 	"https://che-devfile-registry.openshift.io/",
 }
 
+func getDevfileIndexWith(link string, entries *[]DevfileIndexEntry, err error, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	devfileIndexLink := link + "/devfiles/index.json"
+	indexEntries, err := GetDevfileIndex(devfileIndexLink)
+	for i := range indexEntries {
+		indexEntries[i].Links.base = link
+	}
+	*entries = append(*entries, indexEntries...)
```

Comment:
shouldn't we have a mutex lock and unlock here too?

Reply:
Indeed. Missed this one.